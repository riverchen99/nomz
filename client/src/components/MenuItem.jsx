import React from 'react';
import { Link } from 'react-router-dom';
import StarRatingComponent from 'react-star-rating-component';
import { ItemTitle, StarWrapper, StyledMenuItem } from './StyledMenuItem';

function MenuItem(props) {
  const {
    itemName,
    restaurant,
    rating,
    index,
    id
  } = props;
  return (
    <StyledMenuItem index={index}>
      <StarWrapper><StarRatingComponent name={itemName} value={rating} emptyStarColor={"#cccccc"} editing={false} /></StarWrapper>
      <React.Fragment>
        <ItemTitle index={index}>{itemName}</ItemTitle>
        <p>{restaurant}</p>
        <Link to={{ 
          pathname: `/menuitem/${itemName}`, 
          state: {name: itemName, id: id}}}
        >
          Click for details
        </Link>
      </React.Fragment>
    </StyledMenuItem>
  );
}

export default MenuItem;
