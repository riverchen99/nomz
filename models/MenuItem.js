const mongoose = require('mongoose');
// Create Schema

/**
 * @class MenuItem
 */
const MenuItemSchema = new mongoose.Schema({
  _id: { type: String, required: true },
  name: {
    type: String,
    required: true,
  },
  rating: {
    type: Number,
  },
  description: String,
  nutrition: {
    defaultServingSize: Number,
    servingUnit: String,
    calories: Number,
    fatCal: Number,
    totalFat: Number,
    satFat: Number,
    transFat: Number,
    cholesterol: Number,
    sodium: Number,
    totalCarbohydrate: Number,
    dietaryFiber: Number,
    sugars: Number,
    protein: Number,
    vitA: Number,
    vitC: Number,
    calcium: Number,
    iron: Number,
  },
  ingredients: [String],
  allergens: [String],
  props: {
    vegetarian: Boolean,
    vegan: Boolean,
    peanuts: Boolean,
    treeNuts: Boolean,
    wheat: Boolean,
    gluten: Boolean,
    soy: Boolean,
    dairy: Boolean,
    eggs: Boolean,
    shellfish: Boolean,
    fish: Boolean,
    halal: Boolean,
    lowCarbon: Boolean,
  },
  restaurant: { type: mongoose.Schema.Types.ObjectId, ref: 'Restaurant' },
  station: String,
  rating: Number,
});
const MenuItem = mongoose.model('MenuItem', MenuItemSchema);
module.exports = MenuItem;
// note: mongoose automatically uses menuitems as collection name
// https://mongoosejs.com/docs/models.html
