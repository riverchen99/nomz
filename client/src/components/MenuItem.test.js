import React from 'react';
import { shallow, mount, configure } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';
import { MemoryRouter } from 'react-router-dom';
import MenuItem from './MenuItem';

configure({ adapter: new Adapter() });

describe('Menu Item component', () => {
  it('should render correctly without props', () => {
    const component = shallow(<MenuItem />);
    expect(component).toMatchSnapshot();
  });
  it('should render correctly given props', () => {
    const component = mount(<MemoryRouter><MenuItem itemName="bologna" restaurant="covel" rating={2} index={1} id={22} /></MemoryRouter>);
    expect(component).toMatchSnapshot();
  });
  it('has correct Link path to menu item review page', () => {
    const component = mount(<MemoryRouter><MenuItem itemName="bologna" restaurant="covel" rating={2} index={1} id={22} /></MemoryRouter>);
    expect(component.find('Link').props().to.pathname).toEqual('/menuitem/bologna');
    expect(component.find('Link')).toMatchSnapshot();
  });
});
