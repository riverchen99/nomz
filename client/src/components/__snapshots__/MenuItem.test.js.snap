// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Item component has correct Link path to menu item review page 1`] = `ReactWrapper {}`;

exports[`Menu Item component should render correctly given props 1`] = `ReactWrapper {}`;

exports[`Menu Item component should render correctly without props 1`] = `ShallowWrapper {}`;
